torchvtk/augmentation/DictTransform.py

```diff
@@ -0,0 +1,163 @@
+import math
+import random
+from abc import abstractmethod
+from torchvtk.utils.volume_utils import make_5d
+import numpy as np
+import torch
+from torch.nn import functional as F
+
+
+class DictTransform(object):
+    """
+    Super Class for the Transforms.
+
+
+    """
+
+    def __init__(self, device="cpu", apply_on=["vol", "mask"], dtype=torch.float32):
+        """
+        :param device: The torch Code for the device on which the transformation should be executed. Possiblities are ["cpu", "cuda"].
+        :param apply_on: The keys of the volumes that should be transformed.
+        :param dtype: The torch type in which the data are. Possibilities are [torch.float16, torch.float32].
+        """
+        super().__init__()
+
+        self.device = device
+        self.apply_on = apply_on
+        self.dtype = dtype
+
+    @abstractmethod
+    def transform(self, data):
+        """ Transformation Method, must be overwritten by every SubClass."""
+        pass
+
+    def __call__(self, data):
+
+        for key in self.apply_on:
+            tmp = data[key]
+
+            if not torch.is_tensor(tmp):
+                tmp = torch.from_numpy(tmp)
+
+            # to GPU.
+            if self.device == "cuda":
+                tmp = tmp.to(self.device)
+
+            if tmp.dtype is torch.float16:
+                tmp = tmp.to(torch.float32)
+
+            tmp = self.transform(tmp)
+
+            if self.dtype is torch.float16:
+                tmp = tmp.to(torch.float16)
+            elif self.dtype is not torch.float16 or torch.float32:
+                tmp = tmp.to(torch.float32)
+            tmp = tmp.to(self.device)
+            data[key] = tmp
```

Comment:
This can be compacted.
Assuming tmp is a tensor, no checks for device and dtype are necessary since the call to `to` is a noop if the type or device are already correct. 
if tmp is a numpy array its converted to pytorch. Cant do this on `not torch.is_tensor` alone.
if tmp is something completely different, which we might want to support (maybe adding a transform to a transfer function or some keypoints or whatever), just rely on the transform to handle it appropriately (For now those would only be custom implemented transforms, so the user would probably know)

```suggestion
            tmp = data[key]
            isinstance(tmp, np.ndarray): # Convert from NumPy
                tmp = torch.from_numpy(tmp)   
                
            if torch.is_tensor(tmp): # If tmp is tensor, control type and device
                data[key] = self.transform(tmp.to(self.dtype).to(self.device))
            else: 
                data[key] = self.transform(tmp)
```